liltab/data/dataloaders.py

```diff
@@ -15,28 +16,77 @@ class FewShotDataLoader:
 
     def __init__(
         self,
-        dataset: Dataset,
+        dataset: PandasDataset,
         support_size: int,
         query_size: int,
         n_episodes: int = None,
+        sample_classes_equally: bool = False,
+        sample_classes_stratified: bool = False,
     ):
         """
         Args:
-            dataset (Dataset): dataset to load data from.
+            dataset (PandasDataset): dataset to load data from.
             support_size (int): size of support set in each episode.
             query_size (int): size of query set in each episode.
             n_episodes (int, optional): number of episodes.
                 If none, then iterator is without end. Defaults to None.
+            sample_classes_equally (int, optional): If True, then in each iteration gives
+                in task equal number of observations per class.
+                Only makes sense in classification.
+            sample_classes_equally (int, optional): If True, then in each iteration gives
+                in task stratified number of observations per class.
+                Only makes sense in classification.
         """
         self.dataset = dataset
         self.support_size = support_size
         self.query_size = query_size
         self.n_episodes = n_episodes
+        self.sample_classes_equally = sample_classes_equally
+        self.sample_classes_stratified = sample_classes_stratified
+        if self.sample_classes_equally and self.sample_classes_stratified:
+            raise ValueError("Only one of equal or stratified sampling can be used.")
 
         self.curr_episode = 0
 
         self.n_rows = len(self.dataset)
 
+        if sample_classes_equally:
+            self.y = dataset.raw_y
```

Comment:
This step can be done in `datasets` module with one-hot from pytorch (https://pytorch.org/docs/stable/generated/torch.nn.functional.one_hot.html) and reduce memory allocation (both self.y and self.raw_y in datasets)